### PR TITLE
♻️ [amp-selector, amp-accordion] Split component files for packaging

### PR DESCRIPTION
**Pull request by rcebulko**
These ones were a bit messier, but I think I've got the splits right. This PR
- splits component files for `amp-selector`
- splits component files for `amp-accordion`
- defines `PreactBaseElement#triggerEvent`, so Bento components can call into it but AMP component can override/latch on
- defers to `createCustomEvent` within `dispatchCustomEvent`

### CODE REVIEW

File: extensions/amp-selector/1.0/base-element.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+      'as': SelectorShim,
+      'shimDomElement': element,
+      'children': children,
+      'value': value,
+      'options': options,
+    });
+  }
+}
+
+/**
```
Comment (rcebulko) [inline extensions/amp-selector/1.0/base-element.js:93]: Nothing below this line was altered; all cut-pasted directly from `amp-selector.js`

File: extensions/amp-accordion/1.0/base-element.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+} from './component';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {
+  childElementsByTag,
+  dispatchCustomEvent,
+  toggleAttribute,
+} from '../../../src/dom';
+import {dict, memo} from '../../../src/utils/object';
+import {forwardRef} from '../../../src/preact/compat';
+import {getExpandStateTrigger} from './amp-accordion';
```
Comment (jridgewell) [inline extensions/amp-accordion/1.0/base-element.js:32]: We can't import from the amp element in base-element.

File: extensions/amp-accordion/1.0/base-element.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+const HEADER_SHIM_PROP = '__AMP_H_SHIM';
+const CONTENT_SHIM_PROP = '__AMP_C_SHIM';
+const SECTION_POST_RENDER = '__AMP_PR';
+const EXPAND_STATE_SHIM_PROP = '__AMP_EXPAND_STATE_SHIM';
+
+/** @extends {PreactBaseElement<AccordionDef.AccordionApi>} */
+export class BaseElement extends PreactBaseElement {
+  /** @override */
+  init() {
+    this.registerApiAction('toggle', (api, invocation) =>
```
Comment (jridgewell) [inline extensions/amp-accordion/1.0/base-element.js:52]: I believe this is to tie into the AMP event system?

Comment (rcebulko) [reply extensions/amp-accordion/1.0/base-element.js]: Checking the generator template it looks like this indeed should live in the AMP code. Updated.

File: extensions/amp-selector/1.0/base-element.js [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+import * as Preact from '../../../src/preact';
+import {Option, Selector} from './component';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {
+  closestAncestorElementBySelector,
+  createElementWithAttributes,
+  toggleAttribute,
+  tryFocus,
+} from '../../../src/dom';
+import {dev, devAssert, userAssert} from '../../../src/log';
```
Comment (rcebulko) [inline extensions/amp-selector/1.0/base-element.js]: Removing these dependencies...

**Comment by rcebulko** [comment]
bumping @jridgewell , addressed your comments

**Review by jridgewell — APPROVED** [review]

**Pull request merged**